From ffaa454c11524c7fac3d72bca63109f24fac2b7b Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 4 Jul 2024 17:10:01 +0200
Subject: [PATCH] arm64: dts: ti: k3-am642-sr-som: mux ethernet phy reset
 signals input

Specifically on AM64 SoM design, the DP83869 phys have a chance to lock
up if reset gpio are toggled. Update the pinmux to input-only, strongly
enforcing that these signals are left floating at all times.

This avoids sporadic phy initialisation errors mostly encountered during
power-on and reset.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 arch/arm64/boot/dts/ti/k3-am642-sr-som.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am642-sr-som.dtsi b/arch/arm64/boot/dts/ti/k3-am642-sr-som.dtsi
index 238407259..16957a2ae 100644
--- a/arch/arm64/boot/dts/ti/k3-am642-sr-som.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am642-sr-som.dtsi
@@ -379,7 +379,7 @@ AM64X_IOPAD(0x0278, PIN_INPUT, 7)	/* (C19) EXTINTn Mux GPIO1_70 */
 	ethernet_phy0_pins_default: ethernet-phy0-pins-default {
 		pinctrl-single,pins = <
 			/* reset */
-			AM64X_IOPAD(0x0154, PIN_OUTPUT, 7)	/* (V12) PRG1_PRU1_GPO19 Mux GPIO0_84 */
+			AM64X_IOPAD(0x0154, PIN_INPUT, 7)	/* (V12) PRG1_PRU1_GPO19 Mux GPIO0_84 */
 			/* reference clock */
 			AM64X_IOPAD(0x0274, PIN_OUTPUT, 5)	/* (A19) EXT_REFCLK1 Mux CLKOUT0 */
 		>;
@@ -388,7 +388,7 @@ AM64X_IOPAD(0x0274, PIN_OUTPUT, 5)	/* (A19) EXT_REFCLK1 Mux CLKOUT0 */
 	ethernet_phy1_pins_default: ethernet-phy1-pins-default {
 		pinctrl-single,pins = <
 			/* reset */
-			AM64X_IOPAD(0x0150, PIN_OUTPUT, 7)	/* (Y13) PRG1_PRU1_GPO18 Mux GPIO0_20 */
+			AM64X_IOPAD(0x0150, PIN_INPUT, 7)	/* (Y13) PRG1_PRU1_GPO18 Mux GPIO0_20 */
 			/* led0, external pull-down on SoM */
 			AM64X_IOPAD(0x0128, PIN_INPUT, 7)	/* (U12) PRG1_PRU1_GPO8 Mux GPIO0_73 */
 			/* led1/rxer */
@@ -399,7 +399,7 @@ AM64X_IOPAD(0x011c, PIN_INPUT, 7)	/* (AA13) PRG1_PRU1_GPO5 Mux GPIO0_70 */
 	ethernet_phy2_pins_default: ethernet-phy2-pins-default {
 		pinctrl-single,pins = <
 			/* reset */
-			AM64X_IOPAD(0x00d4, PIN_OUTPUT, 7)	/* (U13) PRG1_PRU0_GPO7 Mux GPIO0_52 */
+			AM64X_IOPAD(0x00d4, PIN_INPUT, 7)	/* (U13) PRG1_PRU0_GPO7 Mux GPIO0_52 */
 			/* led0, external pull-down on SoM */
 			AM64X_IOPAD(0x00d8, PIN_INPUT, 7)	/* (W13) PRG1_PRU0_GPO8 Mux GPIO0_53 */
 			/* led1/rxer */
-- 
2.35.3

